tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+        if 'escalated' in caseobj:
+            self.escalated = caseobj['escalated']
+        else:
+            self.escalated = False
+        self.case_creator = caseobj['creator']['displayName']
+        self.create_time = str(datetime.fromisoformat(
+            caseobj['createTime'].replace('Z', '+00:00')))
+        self.update_time = str(datetime.fromisoformat(
+            caseobj['updateTime'].replace('Z', '+00:00')))
+        self.priority = caseobj['severity'].replace('S', 'P')
+        self.state = caseobj['state']
+        req = support_service.cases().comments().list(parent=self.resource_name)
+        self.comment_list = req.execute().get('comments',[])
+
+
+        
+# Handle all calls to the support bot
+@app.route('/google-cloud-support', methods=['POST'])
+def gcp_support() -> Response:
+    """
+    Takes a user's slash command from Slack and executes it. Multiprocessing is used
+    on commands that modify the case to prevent Slack timeouts.
+
+    Parameters
+    ----------
+    request : Request
+        message and metadata that was submitted by Slack
+
+
+    Returns
+    -------
+    Response
+        tells Slack that the command was received and not to throw a timeout alert
+    200
+        HTTP 200 OK
+    """
+    
+    data = request.form
+    token = data.get('token')
+    channel_id = data.get('channel_id')
+    channel_name = data.get('channel_name')
+    user_id = data.get('user_id')
+    user_name = data.get('user_name')
+    user_inputs = data.get('text').split(' ', 1)
+    command = user_inputs[0]
+
+    if command == 'track-case':
+        try:
+            case = user_inputs[1]
+            track_case(channel_id, channel_name, case, user_id)
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text="The track-case command expects argument [case_number]. The case number provided did not match with any cases in your org")
+
+    elif command == 'add-comment':
+        try:
+            parameters = user_inputs[1].split(' ', 1)
+            case = parameters[0]
+            comment = parameters[1]
+            p = Process(target=add_comment, args=(channel_id, case, comment, user_id, user_name,))
+            p.start()
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text="The add-comment command expects arguments [case_number] [comment]. The comment does not need to be encapsulated in quotes. Your case number did not match with any cases in your org.")
+
+    elif command == 'change-priority':
+        try:
+            parameters = user_inputs[1].split(' ', 1)
+            case = parameters[0]
+            priority = parameters[1]
+            p = Process(target=change_priority, args=(channel_id, case, priority, user_id,))
+            p.start()
+        except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.